cocos/3d/CCMesh.cpp

```diff
@@ -613,8 +613,8 @@ void Mesh::setLightUniforms(Pass* pass, Scene* scene, const Vec4& color, unsigne
         if (hasAmbient)
         {
             ambient.x /= 255.f; ambient.y /= 255.f; ambient.z /= 255.f;
+            glProgramState->setUniformVec4("u_color", Vec4(color.x * ambient.x, color.y * ambient.y, color.z * ambient.z, color.w));
```

Comment:
This function is called in Mesh::Draw function, and the function `setUniformVec4("u_color"` will be called in two separate functions, it is better to refactor it into one function.
Thanks.